### ISSUE TITLE
No audio playback in Lowfi

### ISSUE THREAD

**Issue by nfowlie** [talwat/lowfi#17]
When using Lowfi I don't seem to be getting any audio playback.
The application is saying it is playing and volume is set to 100.

I am on Arch using Lowfi v 1.4.0

**Comment by talwat** [comment]
Does it say it's actually playing a song?

I'm pushing an update now that adds a `--debug` flag which you could try to see if it's an issue with ALSA.

**Comment by nfowlie** [comment]
Yeah it says it is playing.
![image](https://github.com/user-attachments/assets/c8daf0b4-17fe-4ffd-aa08-a70b54550352)

**Comment by talwat** [comment]
Try updating to v1.4.1 and sending a screenshot with `lowfi --debug`

**Comment by nfowlie** [comment]
Included in the screenshot is cava.  Showing that there is no audio being played.
![image](https://github.com/user-attachments/assets/40a7e8df-6f23-4b1f-87c0-e747f7611ee6)

**Comment by talwat** [comment]
That is extremely odd. Do you have Alsa installed? Also, do other apps produce audio or is it just lowfi that's the exception?

**Comment by talwat** [comment]
Did an older version of lowfi work properly? 1.4.0 reworked a lot of things so it's possible it might've broke some setups.

**Comment by talwat** [comment]
https://github.com/RustAudio/rodio/issues/173

It could be because of the audio crate I'm using. `lowfi` is tested against pipewire, so maybe you're using pulseaudio..?

**Comment by nfowlie** [comment]
1.4.0 is the first version I have tried.  I just downgraded to 1.3.5 to test and still no audio.

**Comment by nfowlie** [comment]
Yes I am using pulseaudio.  So this only supports pipewire for now?

**Comment by talwat** [comment]
> Yes I am using pulseaudio. So this only supports pipewire for now?

I presume so, unfortunately. I'm using an audio library for rust called rodio, and so if they don't support pulseaudio then lowfi can't either.

Although, as a last possible resort, I could try and compile a build with JACK instead of alsa and seeing if that has playback.

**Comment by talwat** [comment]
Update: On arch the package is called `pulseaudio-alsa`, so try that out and see if it works.

**Comment by nfowlie** [comment]
Yup that was it.
Installing pulseaudio-alsa worked.  I kind of thought I already had that installed.

Thank you very much.  This should probably be added to the documentation.

**Comment by talwat** [comment]
Yep, I'm not very well-versed with Linux audio ironically which is why a lot of these weren't mentioned. Thanks!
